contrib/flo-extras/src/main/java/com/spotify/flo/contrib/dataflow/PipelineOptionsSupplier.java

```diff
@@ -0,0 +1,108 @@
+/*-
+ * -\-\-
+ * Flo Extras
+ * --
+ * Copyright (C) 2016 - 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.flo.contrib.dataflow;
+
+import io.norberg.automatter.AutoMatter;
+import java.util.Optional;
+import java.util.function.Supplier;
+import org.apache.beam.runners.dataflow.DataflowRunner;
+import org.apache.beam.runners.dataflow.options.DataflowPipelineOptions;
+import org.apache.beam.runners.dataflow.options.DataflowPipelineWorkerPoolOptions.AutoscalingAlgorithmType;
+import org.apache.beam.sdk.PipelineRunner;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+
+@AutoMatter
+public interface PipelineOptionsSupplier extends Supplier<DataflowPipelineOptions> {
+
+  String project();
+
+  Optional<Integer> numWorkers();
+
+  Optional<Integer> maxNumWorkers();
+
+  Optional<String> region();
+
+  Optional<String> zone();
+
+  Optional<String> stagingLocation();
+
+  Optional<String> tempLocation();
+
+  Optional<String> gcpTempLocation();
+
+  Optional<AutoscalingAlgorithmType> autoscalingAlgorithm();
```

Comment:
Shall we remove the `Optional` out of the fields that have default values? `maxNumWorkers` `network` `autoscalingAlgorithm` and `runner`. I initially made all optional for no specific reason. I think this should be revisited.

Reply:
Some others might need to be required too (even without having a default value) such as `serviceAccount`?

Reply:
Actually everything is optional except `project` in this case. Users can bypass `.builder()` and start from an empty builder and in that case we should perhaps still make them optional.